src/rabbitmq/RabbitmqClient.ts

```diff
@@ -83,6 +83,7 @@ export abstract class RabbitmqClient {
 
   private async createConnection() {
     const newConnection = await this.connectFunction(this.clientConfig);
+    newConnection.on('blocked', () => { this.handleErrorOrClose('Connection Blocked'); });
```

Comment:
Blocked events are sent when the broker is low on resources(memory, disk). Connection recovery on client side will not ease the issue from the server. A simple handler could be logging the error.